### PR TITLE
Loop parameter leak into for-else block

### LINKED ISSUE #482: Loop parameter leak into for-else block
When a for loop has an [else](https://quarkus.io/guides/qute-reference#loop_section) bock, the item parameter leaks into that else block, it shouldn't be accessible.
```
{#for item in items}
  {item.name}
{#else}
    {ite|}
  No items.
{/for}
```
Screenshot 2021-12-13 at 12 39 21

Screenshot 2021-12-13 at 12 36 46

### PR DESCRIPTION
**Pull request by angelozerr**
Loop parameter leak into for-else block

Fixes #482

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
It starts working, need to write tests:

ElseInForDemo

I wonder if metadata like item_count should be ignored in else block like item alias.

**Comment by fbricon** [comment]
> I wonder if metadata like item_count should be ignored in else\ block like item alias.

In that else block, you get : 
> io.quarkus.qute.TemplateException: Entry "item_count" not found in the data map in expression {item_count} in template items on line 10

**Comment by fbricon** [comment]
Loop metadata should not be proposed in that else block:
Screenshot 2021-12-15 at 19 26 54

**Comment by angelozerr** [comment]
> Loop metadata should not be proposed in that else block:

fixed.

This PR too highlight for nested block. See the following demo:

NestedBlockDemo

It's helpful when you write a #else for #for and for #if and you need to see quickly where the #else belongs to.

I did that too for #switch and #case

**Comment by angelozerr** [comment]
@fbricon you can play with the PR. I need to write tests now.

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/parser/template/sections/BaseWhenSection.java [hunk 1]
```diff
@@ -46,4 +51,8 @@ public ParametersInfo getParametersInfo() {
 		return PARAMETER_INFOS;
 	}
 
+	@Override
+	public List<SectionKind> getBlockLabels() {
+		return ImmutableList.of(SectionKind.IS, SectionKind.CASE, SectionKind.ELSE);
```
Comment (fbricon) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/parser/template/sections/BaseWhenSection.java:56]: use List.of()

Comment (angelozerr) [reply qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/parser/template/sections/BaseWhenSection.java]: thanks fixed

**Comment by angelozerr** [comment]
Tests are written now.

**Comment by fbricon** [comment]
thanks @angelozerr

**Pull request merged**